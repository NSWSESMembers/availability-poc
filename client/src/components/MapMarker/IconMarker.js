import React from 'react';
import PropTypes from 'prop-types';
import { View } from 'react-native';

import Icon from '../Icon';
import styles from './styles';

const IconMarker = ({ name }) => (
  <View>
    <View style={styles.circle} >
      <Icon
        name={name}
        size={20}
        color="white"
        style={{ textAlign: 'center', textAlignVertical: 'center', includeFontPadding: false }}
      />
    </View>
    <View style={styles.arrowBorderBlack} />
    <View style={styles.arrow} />
  </View>
);

IconMarker.propTypes = {
  name: PropTypes.string.isRequired,
};

export default IconMarker;
