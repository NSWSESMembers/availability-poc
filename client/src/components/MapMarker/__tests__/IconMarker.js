import React from 'react';
import renderer from 'react-test-renderer';

import IconMarker from '../IconMarker';

test('renders correctly', () => {
  const tree1 = renderer.create(
    <IconMarker
      name="bullhorn"
    />,
  ).toJSON();
  expect(tree1).toMatchSnapshot();
});
