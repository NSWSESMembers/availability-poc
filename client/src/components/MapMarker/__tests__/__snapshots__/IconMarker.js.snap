// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "black",
        "borderRadius": 50,
        "flex": 1,
        "height": 30,
        "justifyContent": "center",
        "width": 30,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "white",
            "fontSize": 20,
          },
          Object {
            "includeFontPadding": false,
            "textAlign": "center",
            "textAlignVertical": "center",
          },
          Object {
            "fontFamily": "FontAwesome",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
        ]
      }
    >
      ï‚¡
    </Text>
  </View>
  <View
    style={
      Object {
        "alignSelf": "center",
        "backgroundColor": "transparent",
        "borderColor": "transparent",
        "borderTopColor": "black",
        "borderWidth": 4,
        "marginTop": -0.5,
      }
    }
  />
  <View
    style={
      Object {
        "alignSelf": "center",
        "backgroundColor": "transparent",
        "borderColor": "transparent",
        "borderTopColor": "#FF5A5F",
        "borderWidth": 4,
        "marginTop": -9,
      }
    }
  />
</View>
`;
